### DISCUSSION TITLE
what is the meaning of  argument "target" in function get.tree.rfsrc?

### DISCUSSION THREAD

**Discussion by A-Pai** [kogalur/randomForestSRC#237] [Q&A] [answered]
what is the meaning of  argument "target" in function get.tree.rfsrc?
![image](https://user-images.githubusercontent.com/5112397/166643832-11f0e1b2-3bf3-4e11-8100-ca3da70054ad.png)

library(randomForestSRC)

iris.obj <- rfsrc(Species ~., data = iris)
plot(get.tree(iris.obj, 25, target = "versicolor"))
plot(get.tree(iris.obj, 25, target = "virginica"))

![image](https://user-images.githubusercontent.com/5112397/166644056-f16e126b-8bab-4afc-ac90-44cfc6678ba3.png)
![image](https://user-images.githubusercontent.com/5112397/166644132-c6da05b7-d1ef-43d2-acb7-74e6b16198e9.png)
**but it is all the same**

**Comment by ishwaran** [comment, marked as answer]
That's because you are requesting the terminal node value display the class majority since `class.type="bayes"` by default.  This won't change if you focus on a different target.

 See the example in the help file where the goal is to predict probability of class label.

 ```
    ## ------------------------------------------------------------
     ## classification
     ## ------------------------------------------------------------
     
     iris.obj <- rfsrc(Species ~., data = iris, nodesize = 10)
     
     ## equivalent
     plot(get.tree(iris.obj, 25))
     plot(get.tree(iris.obj, 25, class.type = "bayes"))
     
     ## predicted probability displayed for terminal nodes
     plot(get.tree(iris.obj, 25, class.type = "prob", target = "setosa"))
     plot(get.tree(iris.obj, 25, class.type = "prob", target = "versicolor"))
     plot(get.tree(iris.obj, 25, class.type = "prob", target = "virginica"))

 
```
